### PR TITLE
ci: added support for additional cockroachdb tests

### LINKED ISSUE #13268: Add more integration tests for CockroachDB for `db` and `migrate` commands
The prisma engines have good testing of the connector but we would get a better confidence that everything is working as expected with additional integration tests on the CLI side here in prisma/prisma.

Previous issue was about having a minimal testing setup (db pull tests were added)
https://github.com/prisma/prisma/issues/12926

Now we could expand and cover more commands
- [ ] db push (currently, only SQLite is tested: https://github.com/prisma/prisma/issues/13538)
- [ ] migrate reset (currently, only SQLite is tested: https://github.com/prisma/prisma/issues/13539)
- [x] db execute
- [x] migrate dev
- [x] migrate diff

### PR DESCRIPTION
**Pull request by jkomyno**
This PR adds cockroachdb tests for:

- [x] db execute
- [x] migrate dev
- [x] migrate diff 

Closes https://github.com/prisma/prisma/issues/13268

### CODE REVIEW

**Comment by jkomyno** [comment]
Note: merging https://github.com/prisma/prisma/pull/13547 is necessary to make this PR's MacOS tests succeed.

File: packages/migrate/src/__tests__/MigrateResolve.test.ts
Comment (janpio) [inline packages/migrate/src/__tests__/MigrateResolve.test.ts:206]: What causes this unrelated change?

File: packages/migrate/src/__tests__/fixtures/schema-only-cockroachdb/prisma/shadowdb.prisma
Comment (janpio) [inline packages/migrate/src/__tests__/fixtures/schema-only-cockroachdb/prisma/shadowdb.prisma:4]: I see no equivalent renaming of an env var definition anywhere - did I miss it?

**Review by janpio — APPROVED** [review]
Generally looks good, and tests seem to be passing.

(Some actual TS developer should review besides me before merge though)

Also see questions and comments.

**Review by pimeys — APPROVED** [review]
Looks good to me. More tests are better than no tests, especially if they're green. Code quality looks decent to me.

**Review by tomhoule — APPROVED** [review]

File: packages/migrate/src/__tests__/MigrateDev.test.ts
Comment (jkomyno) [reply packages/migrate/src/__tests__/MigrateDev.test.ts:1173]: You're correct. I had simply kept the same test structure as before to minimize the `git diff`

File: packages/migrate/src/__tests__/MigrateResolve.test.ts
Comment (jkomyno) [reply packages/migrate/src/__tests__/MigrateResolve.test.ts:206]: The fact that `packages/migrate/src/__tests__/fixtures/schema-only-cockroachdb/prisma` contains a `.env` file, which is automatically load by the [`loadEnv`](https://github.com/prisma/prisma/blob/c1d4f6ea4a0499759dcc1e43cc6c846ccb8051a8/packages/sdk/src/utils/tryLoadEnvs.ts#L119-L121) function.

File: packages/migrate/src/__tests__/fixtures/schema-only-cockroachdb/prisma/shadowdb.prisma
Comment (janpio) [reply packages/migrate/src/__tests__/fixtures/schema-only-cockroachdb/prisma/shadowdb.prisma:4]: So how did this test ever work? Should it not have failed for a missing env var before? Something doesn't add up, right?

File: packages/migrate/src/__tests__/MigrateDev.test.ts
Comment (janpio) [reply packages/migrate/src/__tests__/MigrateDev.test.ts:1173]: The diff of the snapshot file with all empty strings made me worry this was not testing the right thing, but on checking the test file I assumed it _expected_ those empty strings.

File: packages/migrate/src/__tests__/MigrateResolve.test.ts
Comment (janpio) [reply packages/migrate/src/__tests__/MigrateResolve.test.ts:206]: Ah, and before the `schema-only-cockroachdb` test fixture did not - you added that 👍

File: packages/migrate/src/__tests__/fixtures/schema-only-cockroachdb/prisma/shadowdb.prisma
Comment (jkomyno) [reply packages/migrate/src/__tests__/fixtures/schema-only-cockroachdb/prisma/shadowdb.prisma:4]: The answer is pretty easy: `shadowdb.prisma` was never loaded before. In other words, this pre-existing schema was just noise in the test codebase

File: packages/migrate/src/__tests__/MigrateDev.test.ts
Comment (janpio) [reply packages/migrate/src/__tests__/MigrateDev.test.ts:1173]: You added new snapshots here. The use the suboptimal way to store the snapshots for the use case (= confirming that something does not return actual data). That is what should be changed.

Updating other, existing snapshot logic in other tests should not become part of this PR of course and if we have the same problem elsewhere we should create an issue and fix that independently.

File: packages/migrate/src/__tests__/fixtures/schema-only-cockroachdb/prisma/shadowdb.prisma
Comment (janpio) [reply packages/migrate/src/__tests__/fixtures/schema-only-cockroachdb/prisma/shadowdb.prisma:4]: Ok, that explains it 👍 Good we fixed that then.

File: packages/migrate/src/__tests__/MigrateDev.test.ts
Comment (janpio) [inline packages/migrate/src/__tests__/MigrateDev.test.ts]: Putting this one into an inline snapshot would also be perfectly fine I guess for the tests, no need to have and look into an additional file. But of course not necessary to merge this PR now.

**Review by janpio — APPROVED** [review]

**Pull request merged**
